### PR TITLE
Dan's Suggestions

### PR DESCRIPTION
**Pull request by ddulaney**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,27 +1,9 @@
-FROM gradle:7.1.1-jdk11 as builder
-
 # Build project with gradle
-WORKDIR /workspace
+FROM openjdk:11-jre-slim as builder
```
Comment (ddulaney) [inline Dockerfile:2]: Might as well use the same image to build that you use to run. You're using `./gradlew`, which downloads Gradle, so you're not taking advantage of the Gradle image anyway.

File: Dockerfile [hunk 2]
```diff
@@ -1,27 +1,9 @@
...
 COPY . .
 RUN ./gradlew build
 
-# Store build result as jar file
+# Copy the jar into a new image
 FROM openjdk:11-jre-slim
-ARG uid=1001
-ARG user=interviewee
-
-RUN adduser \
```
Comment (ddulaney) [inline Dockerfile:13]: Running as a non-`root` user is necessary in our k8s clusters, but not really necessary for a practice problem.

File: build.gradle.kts [hunk 3]
```diff
@@ -1,41 +1,41 @@
 import org.jetbrains.kotlin.gradle.tasks.KotlinCompile
 
 plugins {
-	id("org.springframework.boot") version "2.6.0-SNAPSHOT"
```
Comment (ddulaney) [inline build.gradle.kts:4]: You were using a Spring Boot `SNAPSHOT` so it couldn't be downloaded from `mavenCentral()`. I switched to using the latest regular release.

File: build.gradle.kts [hunk 4]
```diff
@@ -1,41 +1,41 @@
...
+
 repositories {
-	mavenCentral()
-	maven { url = uri("https://repo.spring.io/milestone") }
-	maven { url = uri("https://repo.spring.io/snapshot") }
+    mavenCentral()
 }
 
 dependencies {
-	implementation("org.springframework.boot:spring-boot-starter-webflux")
```
Comment (ddulaney) [inline build.gradle.kts:21]: I think `webflux` will cause unnecessary confusion. Assuming they choose to use JDBC or JPA, it wouldn't really be useful anyway.

File: docker-compose.yml [hunk 5]
```diff
@@ -1,209 +1,33 @@
...
-      ZOOKEEPER_TICK_TIME: 2000
-
-  broker:
-    image: confluentinc/cp-server:6.2.0
-    hostname: broker
-    container_name: broker
-    depends_on:
-      - zookeeper
+  kafka:
+    image: lensesio/fast-data-dev
```
Comment (ddulaney) [inline docker-compose.yml:25]: [Fast Data Dev](https://github.com/lensesio/fast-data-dev) provides Kafka, Zookeeper, etc, all in one image.

File: gradlew [hunk 6]
```diff
@@ -1,7 +1,7 @@
-#!/usr/bin/env sh
+#!/bin/sh
```
Comment (ddulaney) [inline gradlew:1]: When I regenerated the project using Spring Initializr for the latest regular release of Spring Boot, it changed the Gradle scripts.

File: src/main/resources/application.yaml [hunk 7]
```diff
@@ -1,23 +0,0 @@
-management:
```
Comment (ddulaney) [inline src/main/resources/application.yaml:1]: I removed Actuator. I don't think it adds any value to the exercise.

File: build.gradle.kts (hunk 3 above)
Comment (nvta-sbiyyala) [reply build.gradle.kts:4]: cool!

File: build.gradle.kts (hunk 4 above)
Comment (nvta-sbiyyala) [reply build.gradle.kts:21]: 👍

File: docker-compose.yml (hunk 5 above)
Comment (nvta-sbiyyala) [reply docker-compose.yml:25]: 👍

File: Dockerfile (hunk 1 above)
Comment (nvta-sbiyyala) [reply Dockerfile:2]: yep!

File: Dockerfile (hunk 2 above)
Comment (nvta-sbiyyala) [reply Dockerfile:13]: sure

File: src/main/resources/application.yaml (hunk 7 above)
Comment (nvta-sbiyyala) [reply src/main/resources/application.yaml:1]: yes, makes sense

**Review by nvta-sbiyyala — APPROVED** [review]

**Pull request merged**